### PR TITLE
Added the steps wrapper modal window

### LINKED ISSUE #137: (SP: 1) Closing confirmation pop-up window for the Stepper
## Description
Implement the closing confirmation pop-up window as it was implemented for **SignUp**. 

## Design 
Image

## Acceptance Criteria
- [ ] If a user clicks on the close button in the Stepper the pop-up window must appear.
- [ ] The pop-up window must not appear if the user clicks outside the Stepper.
- [ ] The pop-up window must disappear if user click on the area of the Stepper.

### PR DESCRIPTION
**Pull request by holus-bolus**
Added the confirmation window similar to the Signupform

2024-05-0113-27-58-ezgif com-video-to-gif-converter

image

### CODE REVIEW

**Review by khytrykv — APPROVED** [review]

**Review by OlhaOvsiienko — APPROVED** [review]

**Comment by voloshkin-v** [comment]
ModalContext already exists, please take a look: https://github.com/Space2Study-UA-1156/Client-01/blob/develop/src/context/modal-context.jsx

app

It seems like to make it work, we need to add `setNeedConfirmation(true)` inside the useEffect.

users-step-wrapper

**Review by voloshynvv — COMMENTED** [review]
ModalContext already exists, and entire app is wrapped inside ModalProvider, please take a look: https://github.com/Space2Study-UA-1156/Client-01/blob/develop/src/context/modal-context.jsx

app

**It seems like to make it work, we just need to add `setNeedConfirmation(true)` inside the useEffect.**

users-step-wrapper

**Comment by holus-bolus** [comment]
@khytrykv , yep, you are correct

**Comment by khytrykv** [comment]
> @khytrykv , yep, you are correct

Not me, but @voloshkin-v

**Review by YaroslavChuiko — APPROVED** [review]

File: src/components/user-steps-wrapper/UserStepsWrapper.jsx
Comment (vlad-khrychov) [inline src/components/user-steps-wrapper/UserStepsWrapper.jsx:1]: Implementation seems well, but you should add unit test for new functional.

**Review by vlad-khrychov — COMMENTED** [review]
Discuss with the team. Maybe we can merge without a test and write the test in another PR for non-blocking development of other tasks.

**Pull request merged**